packages/nivo-pie/index.d.ts

```diff
@@ -0,0 +1,96 @@
+declare module '@nivo/pie' {
+    export class Pie extends React.Component<IPieProps & IDimensions>{}
+    export class ResponsivePie extends React.Component<IPieProps>{}
+
+    export type LegendItemDirection =
+    | 'left-to-right'
+    | 'right-to-left'
+    | 'top-to-bottom'
+    | 'bottom-to-top';
+
+    export type Anchor =
+        | 'top'
+        | 'top-right'
+        | 'right'
+        | 'bottom-right'
+        | 'bottom'
+        | 'bottom-left'
+        | 'left'
+        | 'top-left'
+        | 'center';
+
+    export type LegendDirection = 'row' | 'column';
+
+    export interface INivoLegendItem {
+
+        data: Array<{ label: string | number; fill: string; }>;
+
+        // position & layout
+        anchor: Anchor;
+
+        translateX?: number;
+        translateY?: number;
+        direction: LegendDirection;
+
+        // items
+        itemWidth: number;
+        itemHeight: number;
+        itemDirection?: LegendItemDirection;
+        itemsSpacing?: number;
+        symbolSize?: number;
+        symbolSpacing?: number;
+        symbolShape: "circle" | "diamond" | "square" | "triangle", //Proptypes specifies this as string | func but source implements it as string only.
+        textColor?: string;
+    }
+    
+    export interface IPieDataItem {
+        id: string;
+        value: string;
+    }
+
+    export type SettingsGetterFunc = (dataSlize: IPieDataITem) => string;
+
+    export interface IPieProps {
+        data: Array<IPieDataItem>;
+        sortByValue: bool;
```

Comment:
I just saw that they were all .required by the ReactProps.
Partial is a fine technique by me if they are all optional.

Reply:
Yes, some properties are required, but because the definition targets exposed API, you'll have defaultProps applied, so we have to take propTypes & defaultProps in account when creating definitions.